integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DeltaParser.java

```diff
@@ -0,0 +1,68 @@
+package io.openlineage.spark3.agent.lifecycle.plan.catalog;
+
+import io.openlineage.spark.agent.util.DatasetIdentifier;
+import io.openlineage.spark.agent.util.PathUtils;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.TableIdentifier;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.delta.catalog.DeltaCatalog;
+import scala.Option;
+
+@Slf4j
+public class DeltaParser implements CatalogParser {
+  public boolean hasClasses() {
+    try {
+      IcebergParser.class
+          .getClassLoader()
+          .loadClass("org.apache.spark.sql.delta.catalog.DeltaCatalog");
+      return true;
+    } catch (Exception e) {
+      // swallow- we don't care
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isClass(TableCatalog tableCatalog) {
+    return tableCatalog instanceof DeltaCatalog;
+  }
+
+  @Override
+  public DatasetIdentifier getDatasetIdentifier(
+      SparkSession session,
+      TableCatalog tableCatalog,
+      Identifier identifier,
+      Map<String, String> properties) {
+    DeltaCatalog catalog = (DeltaCatalog) tableCatalog;
+
+    Optional<String> location;
+    if (catalog.isPathIdentifier(identifier)) {
+      location = Optional.of(identifier.name());
+    } else {
+      location = Optional.ofNullable(properties.get("location"));
+    }
+    // Delta uses spark2 catalog when location isn't specified.
+    Path path =
+        new Path(
+            location.orElse(
```

Comment:
Maybe I'm blind, but I don't see how that will make this one better?

Reply:
`orElseGet` calls a supplier function, which is only invoked in the case that the location isn't present. `orElse` forces you to actually execute the table lookup from the catalog, which could end up causing exceptions to be thrown or some other undesirable side effect, even if the location is present and you don't actually need the return value from the catalog. I think it's better to avoid executing method calls that are unnecessary and using a supplier function is a way to accomplish that.